logstash-core/lib/logstash/api/lib/app/service.rb

```diff
@@ -40,4 +44,15 @@ def get(key)
     end
     LogStash::Json.dump(data)
   end
+
+  private
+
+  def has_counters?
+    return false if @snapshot.nil?
+    if @snapshot_rotation_mutex.try_lock
+      has = @snapshot.metric_store.all.clone.map { |t| t.class.to_s }.include?("LogStash::Instrument::MetricType::Counter")
+      @snapshot_rotation_mutex.unlock
+      return has
+    end
+  end
```

Comment:
that would help, but my issue here is more on the side of having counters there. it takes some time for the pipeline to actually generate some events acouting information, and to test for example the  `/events` resource I do need this to be there. JVM periodic pollers is working like charm and providing data very fast.